packages/nc-gui/components/import/JSONImport.vue

```diff
@@ -0,0 +1,479 @@
+      this.loadUrl()
+    }
+  },
+  methods: {
+    formatJson() {
+      console.log(this.$refs.editor)
+      this.$refs.editor.format()
+    },
+
+    selectFile() {
+      this.$refs.file.files = null
+      this.$refs.file.click()
+    },
+
+    _change(event) {
+      const files = event.target.files
+      if (files && files[0]) {
+        this._file(files[0])
+        event.target.value = ''
+      }
+    },
+    async _file(file) {
+      this.templateData = null
+      this.importData = null
+      this.$store.commit('loader/MutMessage', 'Loading excel file')
+      let i = 0
+      const int = setInterval(() => {
+        this.$store.commit('loader/MutMessage', `Loading excel file${'.'.repeat(++i % 4)}`)
+      }, 1000)
+      this.dropOrUpload = false
+      const reader = new FileReader()
+      this.filename = file.name
+
+      reader.onload = async(e) => {
+        const ab = e.target.result
+        await this.parseAndExtractData('file', ab, file.name)
+        this.$store.commit('loader/MutMessage', null)
+
+        clearInterval(int)
+      }
+
+      const handleEvent = (event) => {
+        this.$store.commit('loader/MutMessage', `${event.type}: ${event.loaded} bytes transferred`)
+      }
+
+      reader.addEventListener('progress', handleEvent)
+      reader.onerror = (e) => {
+        console.log('error', e)
+        this.$store.commit('loader/MutClear')
+      }
+      reader.readAsText(file)
+    },
+
+    async parseAndExtractData(type, val, name) {
+      try {
+        let templateGenerator
+        this.templateData = null
+        this.importData = null
+        switch (type) {
+          case 'file':
+            templateGenerator = new JSONTemplateAdapter('JSON', val, this.parserConfig)
+            break
+          case 'url':
+            templateGenerator = new JSONUrlTemplateAdapter(val, this.$store, this.parserConfig, this.$api)
+            break
+          case 'string':
+            templateGenerator = new JSONTemplateAdapter('JSON', val, this.parserConfig)
+            break
+        }
+        await templateGenerator.init()
+        templateGenerator.parse()
+        this.templateData = templateGenerator.getTemplate()
+
+        this.templateData.tables[0].table_name = this.populateUniqueTableName()
+
+        this.importData = templateGenerator.getData()
+        this.templateEditorModal = true
+      } catch (e) {
+        console.log(e)
+        this.$toast
+          .error(await this._extractSdkResponseErrorMsg(e))
+          .goAway(3000)
+      }
+    },
+
+    dropHandler(ev) {
+      this.dragOver = false
+      let file
+      if (ev.dataTransfer.items) {
+        // Use DataTransferItemList interface to access the file(s)
+        if (ev.dataTransfer.items.length && ev.dataTransfer.items[0].kind === 'file') {
+          file = ev.dataTransfer.items[0].getAsFile()
+        }
+      } else if (ev.dataTransfer.files.length) {
+        file = ev.dataTransfer.files[0]
+      }
+
```

Comment:
Suggested to check if the inputting file is a json here. If not, return error message using `this.$toast.error(...)`. Just like other import types.